src/compiler/crystal/syntax/location.cr

```diff
@@ -1,13 +1,12 @@
-# A location of an `ASTnode`, including its filename, line number and column number.
-class Crystal::Location
-  include Comparable(self)
-
-  getter line_number
-  getter column_number
-  getter filename
+require "./virtual_file"
 
-  def initialize(@filename : String | VirtualFile | Nil, @line_number : Int32, @column_number : Int32)
-  end
+# A location of an `ASTNode`, including its filename, line number, column number and (optional) size.
+record Crystal::Location,
+  filename : String | VirtualFile | Nil,
+  line_number : Int32,
```

Comment:
Should these be `Int64`?
I had some problems with large files but I am not sure if this would matter.

Reply:
I don't think a Crystal file with more than Int32::MAX lines or columns is feasible.

Reply:
I was looking at the suspected file and you are correct.  I still don't know what limit in the compiler stopped the compiler from working.